src/compiler/checker.ts

```diff
@@ -17501,8 +17501,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
 
     function isDistributionDependent(root: ConditionalRoot) {
         return root.isDistributive && (
-            isTypeParameterPossiblyReferenced(root.checkType as TypeParameter, root.node.trueType) ||
-            isTypeParameterPossiblyReferenced(root.checkType as TypeParameter, root.node.falseType));
+            containsReference(root.checkType as TypeParameter, root.node.trueType) ||
+            containsReference(root.checkType as TypeParameter, root.node.falseType));

```

Comment:
The new logic more closely matches the comment at the only call site of [`isDistributionDependent`](https://github.dev/microsoft/TypeScript/blob/68df4befd68c0325861bd5b87c6e4d251ffd5a41/src/compiler/checker.ts#L21010-L21012):
> // We check for a relationship to a conditional type target only when the conditional type has no
> // 'infer' positions and is not distributive or is distributive but doesn't reference the check type
> // parameter in either of the result types.